TypeCobol/Compiler/Scanner/Scanner.cs

```diff
@@ -540,22 +540,23 @@ public static void ScanTokensLineContinuationGroup(IList<TokensLine> continuatio
         }
 
         /// <summary>
-        /// Scan an isolated token in the following "default" context :
+        /// Scan an isolated token given a a Scan State.
+        /// If the given scan state is null then Scan in performed in the following "default" context :
         /// - insideDataDivision = true
         /// - decimalPointIsComma = false
         /// - withDebuggingMode = false
         /// - encodingForAlphanumericLiterals = IBM 1147
         /// - default compiler options
         /// </summary>
-        public static Token ScanIsolatedTokenInDefaultContext(string tokenText, out Diagnostic error)
+        public static Token ScanIsolatedTokenInDefaultContext(MultilineScanState scanState, string tokenText, out Diagnostic error)
```

Comment:
This does not fit with the name of the method, the purpose of `ScanIsolatedTokenInDefaultContext` is to provide a default context so the caller does not have to consider the scan state. So maybe it is better to create a new method `ScanIsolatedToken` which accepts a scan state parameter (as a bonus, ScanIsolatedTokenInDefaultContext will be able to call this new method providing its own default scan state).

Reply:
I agree too, but for me after this PR, ScanIsolatedTokenInDefaultContext must not be used anymore.
As the problem is that we always need a context to scan token.

Reply:
@smedilol is right, this is the only usage of the method, so renaming it is enough.

Reply:
So the idea here is to create a private method, and use two methods with distinct names that call the private one. But the call to call the methods is performed at only one place, thus I prefer rename it: for instance ScanIsolatedTokenInGivenContextorDefault